Add sync/force sync behavior to monitors/SLOs/Dashboards docs (#2199)
DataDog/datadog-operator: docs/datadog_monitor.md

    Status:                True
    Type:                  Active
    Last Transition Time:  2025-09-22T19:10:26Z
    Last Update Time:      2025-09-22T19:10:26Z
    Message:               DatadogMonitor Updated
    Status:                True
    Type:                  Updated
  Created:                 2025-09-22T19:10:21Z
  Creator:                 eman.okyere@datadoghq.com
  Current Hash:            009148d415bf6e16d2f920e69d03a5f3
  Downtime Status:
  Id:                                  217179025
  Monitor Last Force Sync Time:        2025-09-22T19:10:26Z
  Monitor State:                       OK
  Monitor State Last Transition Time:  2025-09-22T19:11:27Z
  Monitor State Last Update Time:      2025-09-22T19:32:31Z
  Monitor State Sync Status:           OK
  Primary:                             true
Events:
  Type    Reason                 Age   From            Message
  ----    ------                 ----  ----            -------
  Normal  Create DatadogMonitor  22m   DatadogMonitor  system/datadog-event-v2-alert-test
```
Unlike dashboards and SLOs, Monitor state is synced every minute to ensure that changes to state are tracked in K8s. 

> This means that if a monitor's state transitions from OK to Warn, the CR's state will be updated to Warn in a minute. 

Reviewer: ```suggestion
This means that if a monitor's state transitions from OK to Warn, the CR's state gets updated to Warn in a minute. 
```
